### PR TITLE
Add listing and playing recorded interactive sessions to tsh docs

### LINKED ISSUE #4714: Missing section on replaying session recordings in user manual
## Summary
In the ["Using tsh"](https://goteleport.com/docs/use-teleport/tsh/) guide, we promise users will learn how to:

- Replay recorded interactive sessions.

But we don't have a corresponding section that actually says how to do that.

### PR DESCRIPTION
**Pull request by stevenGravy**

### CODE REVIEW

File: docs/pages/cspell.json [hunk 1]
```diff
@@ -0,0 +1,613 @@
+// cSpell Settings
```
Comment (zmb3) [inline docs/pages/cspell.json:1]: Was this added intentionally? I'd recommend a separate PR if you want to enable spellchecking.

File: docs/pages/connect-your-client/tsh.mdx [hunk 2]
```diff
@@ -842,5 +842,59 @@ DEBU [TSH]       Self re-exec command: tsh [status --format=json]. tsh/aliases.g
 ...
 ```
 
+# Interactive Recording
+
+Within the Teleport cluster the `ssh` and `Kubernetes` recorded sessions are available for listing and replay for authorized users via `tsh`.
+The same listings including Desktops are available in the Web Console under the Audit Log in the Activity section.
```
Comment (zmb3) [inline docs/pages/connect-your-client/tsh.mdx:848]: Recordings are listed under "Session Recordings" - not under audit log, right?

File: docs/pages/connect-your-client/tsh.mdx [hunk 3]
```diff
@@ -842,5 +842,59 @@ DEBU [TSH]       Self re-exec command: tsh [status --format=json]. tsh/aliases.g
 ...
 ```
 
+# Interactive Recording
+
+Within the Teleport cluster the `ssh` and `Kubernetes` recorded sessions are available for listing and replay for authorized users via `tsh`.
```
Comment (zmb3) [inline docs/pages/connect-your-client/tsh.mdx:847]: I don't think you need to code-block `ssh` or `Kubernetes` here. Just say "SSH and Kubernetes sessions"


Also, this sentence is pretty long. Maybe we can simplify:

> Recordings for SSH and Kubernetes sessions can be listed with the `tsh` tool.

File: docs/pages/connect-your-client/tsh.mdx [hunk 4]
```diff
@@ -842,5 +842,59 @@ DEBU [TSH]       Self re-exec command: tsh [status --format=json]. tsh/aliases.g
...
+The same listings including Desktops are available in the Web Console under the Audit Log in the Activity section.
+
+## List Recordings and Play
+
+Get the list of recordings.  Without specifying a time period the past 24 hour recordings are shown.
```
Comment (zmb3) [inline docs/pages/connect-your-client/tsh.mdx:852]:
```suggestion
Get the list of recordings. By default, recordings for the past 24 hours are shown.
The time range can be customized with the `--last`, `--from-utc`, and `--to-utc` flags.
```

File: docs/pages/connect-your-client/tsh.mdx [hunk 5]
```diff
@@ -842,5 +842,59 @@ DEBU [TSH]       Self re-exec command: tsh [status --format=json]. tsh/aliases.g
...
+```code
+$ tsh recordings ls
+ID                                   Type Participants Hostname Timestamp           
+------------------------------------ ---- ------------ -------- ------------------- 
+b0a04442-70dc-4be8-9308-7b7901d2d600      jeff         dev       Nov 26 16:36:16 UTC 
+c0a02222-70dc-4be8-9308-7b7901d2d600 kube alice                  Nov 26 20:36:16 UTC 
+d0a04442-70dc-4be8-9308-7b7901d2d600      navin        test      Nov 26 16:36:16 UTC 
+```
+
+If no type is specified that is a `ssh` session.
```
Comment (zmb3) [inline docs/pages/connect-your-client/tsh.mdx:862]: This seems like an oversight. We should fix this in the code.

File: docs/pages/connect-your-client/tsh.mdx [hunk 6]
```diff
@@ -842,5 +842,59 @@ DEBU [TSH]       Self re-exec command: tsh [status --format=json]. tsh/aliases.g
...
+
+To Play sessions type `tsh play <sessionid>`.  The screen will reset playing the interactive session with the timestamp in the upper right.
```
Comment (zmb3) [inline docs/pages/connect-your-client/tsh.mdx:864]:
```suggestion
To play sessions, use the `tsh play <sessionid>` command. The screen will reset playing the interactive session with the timestamp in the upper right.
```

File: docs/pages/connect-your-client/tsh.mdx [hunk 7]
```diff
@@ -842,5 +842,59 @@ DEBU [TSH]       Self re-exec command: tsh [status --format=json]. tsh/aliases.g
...
+root@ubuntu:/# exit
+exit
+
+
+end of session playback
+```
+
+### Filtering Recorded sessions
+
+Multiple options are available to get recordings from specific days.  The `--from-utc` and `--to-utc` parameters
```
Comment (zmb3) [inline docs/pages/connect-your-client/tsh.mdx:887]: Don't forget the `--last` flag.

But on second thought - this is a pretty high level page. I would keep this page brief and move a more detailed reference to docs/pages/reference/cli.mdx.

File: docs/pages/cspell.json (hunk 1 above)
Comment (stevenGravy) [reply docs/pages/cspell.json:1]: No will remove, thanks!

File: docs/pages/connect-your-client/tsh.mdx (hunk 3 above)
Comment (alexfornuto) [reply docs/pages/connect-your-client/tsh.mdx:847]: Agreed re: code blocks. Additionally, aren't we replacing the term "SSH Access" with "Server Access"?

File: docs/pages/connect-your-client/tsh.mdx (hunk 2 above)
Comment (alexfornuto) [reply docs/pages/connect-your-client/tsh.mdx:848]:
```suggestion
The same listings, including Desktop Access sessions, are available in the Web Console under the Audit Log in the Activity section.
```

File: docs/pages/connect-your-client/tsh.mdx [hunk 8]
```diff
@@ -842,5 +842,60 @@ DEBU [TSH]       Self re-exec command: tsh [status --format=json]. tsh/aliases.g
...
 ```
 
+# Interactive Recording
+
+Within the Teleport cluster the `ssh` and `Kubernetes` recorded sessions are available for listing and replay for authorized users via `tsh`.
+The same listings including Desktops are available in the Web Console under the Audit Log in the Activity section.
+
+## List Recordings and Play
+
+Get the list of recordings. By default, recordings for the past 24 hours are shown.
```
Comment (alexfornuto) [inline docs/pages/connect-your-client/tsh.mdx:852]:
```suggestion
By default, recordings for the past 24 hours are shown.
```

Not sure what value that first sentence added.

File: docs/pages/connect-your-client/tsh.mdx [hunk 9]
```diff
@@ -842,5 +842,60 @@ DEBU [TSH]       Self re-exec command: tsh [status --format=json]. tsh/aliases.g
 ...
 ```
 
+# Interactive Recording
+
+Within the Teleport cluster the `ssh` and `Kubernetes` recorded sessions are available for listing and replay for authorized users via `tsh`.
+The same listings including Desktops are available in the Web Console under the Audit Log in the Activity section.
+
+## List Recordings and Play
```
Comment (alexfornuto) [inline docs/pages/connect-your-client/tsh.mdx:850]:
```suggestion
## List and Play Recordings
```

File: docs/pages/connect-your-client/tsh.mdx [hunk 10]
```diff
@@ -842,5 +842,60 @@ DEBU [TSH]       Self re-exec command: tsh [status --format=json]. tsh/aliases.g
...
+
+Get the list of recordings. By default, recordings for the past 24 hours are shown.
+The time range can be customized with the `--last`, `--from-utc`, and `--to-utc` flags.
+```code
```
Comment (alexfornuto) [inline docs/pages/connect-your-client/tsh.mdx:854]: Code blocks should have surrounding newlines.

```suggestion
The time range can be customized with the `--last`, `--from-utc`, and `--to-utc` flags.

```code
```

**Comment by alexfornuto** [comment]
Should we mention something about `tsh recordings ls` listing recordings that are not accessible depending on your storage configuration? For example, with just the default local storage settings:

```
❯ tsh recordings ls --from-utc 2022-11-20
ID                                   Type Participants     Hostname          Timestamp
------------------------------------ ---- ---------------- ----------------- -------------------
e3d10ea9-26f6-403d-aed6-ba83c47c4845      alex@fornuto.com xeon.tp.alexf.xyz Nov 29 20:25:36 UTC
c8e1b2c5-322a-4095-89e3-391edfd2da9b      alex@fornuto.com xeon.tp.alexf.xyz Nov 28 18:42:24 UTC
648bb0d1-a646-49e3-8bc0-3d8023ef54f6      alex@fornuto.com xeon.tp.alexf.xyz Nov 28 18:28:31 UTC
...
...

❯ tsh play e3d10ea9-26f6-403d-aed6-ba83c47c4845
# The session recoding from today plays

❯ tsh play c8e1b2c5-322a-4095-89e3-391edfd2da9b
ERROR: Recording for session c8e1b2c5-322a-4095-89e3-391edfd2da9b not found.

❯ tsh play 648bb0d1-a646-49e3-8bc0-3d8023ef54f6
ERROR: Recording for session 648bb0d1-a646-49e3-8bc0-3d8023ef54f6 not found.
```

I assume the solve here is to unpack the sessions recording from that day.

File: docs/pages/connect-your-client/tsh.mdx (hunk 3 above)
Comment (stevenGravy) [reply docs/pages/connect-your-client/tsh.mdx]: Updated @zmb3 @alexfornuto

File: docs/pages/connect-your-client/tsh.mdx (hunk 5 above)
Comment (stevenGravy) [reply docs/pages/connect-your-client/tsh.mdx]: Addressing this in another PR

File: docs/pages/connect-your-client/tsh.mdx (hunk 7 above)
Comment (stevenGravy) [reply docs/pages/connect-your-client/tsh.mdx]: added these in the CLI reference

**Comment by stevenGravy** [comment]
> tsh play c8e1b2c5-322a-4095-89e3-391edfd2da9b
> ERROR: Recording for session c8e1b2c5-322a-4095-89e3-391edfd2da9b not found.

@alexfornuto please see the cli reference updates.

File: docs/pages/reference/cli.mdx [hunk 11]
```diff
@@ -311,6 +319,30 @@ $ tsh recordings ls --from-utc=2022-11-01 --to-utc=2022-11-3
...
 ```
 
+<Admonition
+  type="note"
+  title="Recorded Sessions Availability"
+  scope={["oss", "enterprise"]} scopeOnly
+>
+
+Recorded sessions are linked from the audit events to session recordings files. Self-hosted high availability Teleport
+deployments with directory storage for audit logs and session recordings can result in unavailable playback. That occurs
```
Comment (alexfornuto) [inline docs/pages/reference/cli.mdx:329]: >Self-hosted high availability Teleport
deployments with directory storage for audit logs and session recordings can result in unavailable playback. 

Can we rephrase? This sentence is long and hard to parse. Not sure how to edit or I'd offer a suggestion.

Worth noting though that in my case the issue wasn't with an HA setup and separate storage locations. Instead I suspect the issue was with rotation and older sessions being archived/compressed.

File: docs/pages/reference/cli.mdx [hunk 12]
```diff
@@ -311,6 +319,30 @@ $ tsh recordings ls --from-utc=2022-11-01 --to-utc=2022-11-3
...
+since each auth server has its own audit log and recorded session records.
+
+```code
+$ tsh play c8e1b2c5-322a-4095-89e3-391edfd2da9b
+ERROR: Recording for session c8e1b2c5-322a-4095-89e3-391edfd2da9b not found.
+```
+
+Using a SIEM that combines the audit logs will help consolidate the list of available recordings. Downloading the
```
Comment (alexfornuto) [inline docs/pages/reference/cli.mdx:337]: Is it safe to assume that all readers of this page will be familiar with the acronym SIEM?

File: docs/pages/reference/cli.mdx [hunk 13]
```diff
@@ -311,6 +319,30 @@ $ tsh recordings ls --from-utc=2022-11-01 --to-utc=2022-11-3
...
+deployments with directory storage for audit logs and session recordings can result in unavailable playback. That occurs
+since each auth server has its own audit log and recorded session records.
+
+```code
+$ tsh play c8e1b2c5-322a-4095-89e3-391edfd2da9b
+ERROR: Recording for session c8e1b2c5-322a-4095-89e3-391edfd2da9b not found.
+```
+
+Using a SIEM that combines the audit logs will help consolidate the list of available recordings. Downloading the 
+recorded session is directly playable as a file.
```
Comment (alexfornuto) [inline docs/pages/reference/cli.mdx:338]:
```suggestion
Using a SIEM that combines the audit logs will help consolidate the list of available recordings. Downloaded recorded session are directly playable as a file. 
```

File: docs/pages/reference/cli.mdx (hunk 12 above)
Comment (stevenGravy) [reply docs/pages/reference/cli.mdx]: provided

File: docs/pages/reference/cli.mdx (hunk 11 above)
Comment (stevenGravy) [reply docs/pages/reference/cli.mdx]: rephrased @alexfornuto

File: docs/pages/reference/cli.mdx [hunk 14]
```diff
@@ -325,16 +325,17 @@ $ tsh recordings ls --last=6h0m0s
   scope={["oss", "enterprise"]} scopeOnly
 >
 
-Recorded sessions are linked from the audit events to session recordings files. Self-hosted high availability Teleport
-deployments with directory storage for audit logs and session recordings can result in unavailable playback. That occurs
-since each auth server has its own audit log and recorded session records.
+Recorded sessions are linked from the audit events to session recordings files in their [storage backend](../reference/backends.mdx).
+The following error can occur if a session recording file is not available or when employing multiple auth servers with directory storage
+backend for recorded sessions. When using directory storage backend for audit logs and recorded sessions then only the auth server with that
+recorded session can retrieve it.
```
Comment (alexfornuto) [inline docs/pages/reference/cli.mdx:331]: Line breaks and minor copy edits.
```suggestion
Recorded sessions are linked from the audit events to session recordings files
in their [storage backend](../reference/backends.mdx).
The following error can occur if a session recording file is not available or
when employing multiple auth servers with directory storage backend for recorded
sessions. When using a directory storage backend for audit logs and recorded sessions,
only the auth server with that recorded session can retrieve it.
```

File: docs/pages/reference/cli.mdx [hunk 15]
```diff
@@ -325,16 +325,17 @@ $ tsh recordings ls --last=6h0m0s
...
 
 ```code
 $ tsh play c8e1b2c5-322a-4095-89e3-391edfd2da9b
 ERROR: Recording for session c8e1b2c5-322a-4095-89e3-391edfd2da9b not found.
 ```
 
-Using a SIEM that combines the audit logs will help consolidate the list of available recordings. Downloaded recorded session are directly playable as a file. 
+Using a Security information and event management (SIEM) service that combines the audit logs will help consolidate the list of available recordings. Downloaded recorded session are directly playable as a file.
```
Comment (alexfornuto) [inline docs/pages/reference/cli.mdx:338]: - Capitalized the words in the acronym
- Put the last sentence in a new paragraph since it begins a new topic.

```suggestion
Using a Security Information and Event Management (SIEM) service that combines
the audit logs will help consolidate the list of available recordings.

Downloaded recorded session are directly playable as a file.
```

**Review by alexfornuto — APPROVED** [review]
👍 pending comments.

**Comment by stevenGravy** [comment]
> 👍 pending comments.

Should be good now

File: docs/pages/connect-your-client/tsh.mdx [hunk 16]
```diff
@@ -842,5 +842,45 @@ DEBU [TSH]       Self re-exec command: tsh [status --format=json]. tsh/aliases.g
 ...
 ```
 
+# Interactive Recording
+
+Within the Teleport cluster the Server and Kubernetes recorded sessions are available for listing and replay for authorized users via `tsh`.
+The same listings, including Desktop Access sessions, are available in the Web Console under the Audit Log in the Activity section.
```
Comment (zmb3) [inline docs/pages/connect-your-client/tsh.mdx:848]:
```suggestion
The same listings, including Desktop Access sessions, are available in the Web Console under Sesison Recordings in the Activity section.
```

File: docs/pages/connect-your-client/tsh.mdx [hunk 17]
```diff
@@ -842,5 +842,45 @@ DEBU [TSH]       Self re-exec command: tsh [status --format=json]. tsh/aliases.g
...
+
+## List and Play Recordings
+
+Get the list of recording sessions. By default, recordings for the past 24 hours are shown.
```
Comment (zmb3) [inline docs/pages/connect-your-client/tsh.mdx:852]:
```suggestion
Get the list of recorded sessions. By default, recordings for the past 24 hours are shown.
```

File: docs/pages/reference/cli.mdx [hunk 18]
```diff
@@ -276,6 +276,78 @@ Run `tsh help <subcommand>` or see the [Global Flags section](#tsh-global-flags)
...
+#### Flags
+
+| Name | Default Value(s) | Allowed Value(s) | Description |
+| - | - | - | - |
+| `--from-utc` | 24 hours ago | date | Start of time range in which recordings are listed. Format 2006-01-02. Defaults to 24 hours ago. |
+| `--to-utc` | current | date | Start of time range in which recordings are listed. Format 2006-01-02. Defaults to 24 hours ago. |
+| `--limit` | 50 | number | Maximum number of recordings to show. |
+| `--last` | none | duration | Duration into the past from which session recordings should be listed. Format 5h30m40s |
+
+#### Global flags
```
Comment (zmb3) [inline docs/pages/reference/cli.mdx:296]: Not a question for this PR, but I think it's silly that we duplicate this section for every single command.

cc @alexfornuto @ptgott

File: docs/pages/reference/cli.mdx [hunk 19]
```diff
@@ -276,6 +276,78 @@ Run `tsh help <subcommand>` or see the [Global Flags section](#tsh-global-flags)
...
+```code
+# get the recorded sessions from the last 24 hours
+$ tsh --proxy proxy.example.com recordings ls
+ID                                   Type Participants Hostname Timestamp
+------------------------------------ ---- ------------ -------- -------------------
+b0a04442-70dc-4be8-9308-7b7901d2d600 ssh  jeff         dev       Nov 26 16:36:16 UTC
+c0a02222-70dc-4be8-9308-7b7901d2d600 kube alice                  Nov 26 20:36:16 UTC
+d0a04442-70dc-4be8-9308-7b7901d2d600 ssh  navin        test      Nov 26 16:36:16 UTC
+
+#After retrieving the recorded session can be played with tsh play
```
Comment (zmb3) [inline docs/pages/reference/cli.mdx:312]:
```suggestion
# The session can be played with tsh play
```

File: docs/pages/reference/cli.mdx [hunk 20]
```diff
@@ -276,6 +276,78 @@ Run `tsh help <subcommand>` or see the [Global Flags section](#tsh-global-flags)
...
+$ tsh play c0a02222-70dc-4be8-9308-7b7901d2d60
+
+# Retrieve recorded sessions that occurred between Nov 1, 2022 to Nov 3, 2022
```
Comment (zmb3) [inline docs/pages/reference/cli.mdx:315]:
```suggestion
# List recorded sessions that occurred between Nov 1, 2022 to Nov 3, 2022
```

"retrieve" sounds like "download" to me. This is just a listing, it's not really "retrieving" anything

File: docs/pages/reference/cli.mdx [hunk 21]
```diff
@@ -276,6 +276,78 @@ Run `tsh help <subcommand>` or see the [Global Flags section](#tsh-global-flags)
...
+$ tsh recordings ls --from-utc=2022-11-01 --to-utc=2022-11-3
+
+# Retrieve recorded sessions in the last 6 hours 
+$ tsh recordings ls --last=6h0m0s
+```
+
+<Admonition
```
Comment (zmb3) [inline docs/pages/reference/cli.mdx:322]: This might make more sense in a troubleshooting guide if we have one.

Comment (stevenGravy) [reply docs/pages/reference/cli.mdx:322]: @alexfornuto thoughts?

Comment (alexfornuto) [reply docs/pages/reference/cli.mdx:322]: We don't have a troubleshooting page for the Reference section. We could:
- Add this into a troubleshooting section for the existing [Audit Events and Records](https://goteleport.com/docs/reference/audit/) page,
- Incorporate it into Zac's [drafted](https://github.com/gravitational/teleport/pull/18717) overview of session recording,
- Just keep it here.

File: docs/pages/reference/cli.mdx (hunk 18 above)
Comment (alexfornuto) [reply docs/pages/reference/cli.mdx:296]: Agreed. I haven't been around long enough to know what convention or conversation (if any) led to this being the case, but I think it's worth revisiting.

File: docs/pages/reference/cli.mdx (hunk 21 above)
Comment (stevenGravy) [reply docs/pages/reference/cli.mdx:322]: @alexfornuto I would just leave it in here

Comment (alexfornuto) [reply docs/pages/reference/cli.mdx:323]: Fine by me. Consider using a [type](http://localhost:3000/docs/contributing/documentation/reference/#admonitions) other than `note`.

**Review by strideynet — APPROVED** [review]
